dev/tasks/r/github.packages.yml

```diff
@@ -375,29 +375,30 @@ jobs:
           read_parquet(system.file("v0.7.1.parquet", package = "arrow"))
           print(arrow_info())
 
-  #TODO test macos source build?
-  test-linux-source:
+  test-source:
     needs: source
-    name: Test linux source build
-    runs-on: ubuntu-latest
+    name: Test {{ '${{ matrix.platform.name }}' }} source build
+    runs-on: {{ '${{ matrix.platform.runs_on }}' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          - {runs_on: "ubuntu-latest", name: "Linux"}
+          - {runs_on: ["self-hosted", "macos-10.13"] , name: "macOS"}

```

Comment:
It might be nice to add a comment here that explains what `self-hosted` is. It's an M* mac with macOS 14 on it? Or 13? Or? Or, if this is documented somewhere else that's public link to that so that folks can figure out what this is?

Reply:
This is just one machine, the 10.13 runner but self-hosted runners can be identified by multiple tags that are group in an array. For example we have both linux and macos arm64 runners so we have to use `["self-hosted", "arm64", "macos"]` for the m1. Adding "self-hosted" is recommended by gh and the default tag all self-hosted runners have to avoid any collision with official runners. See https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/using-self-hosted-runners-in-a-workflow#using-default-labels-to-route-jobs for details.

Reply:
Aaaah ok, got it. Thanks